File: app/src/main/java/com/akay/fitnass/view/activities/DetailActivity.java
package com.akay.fitnass.view.activities;

import android.arch.lifecycle.ViewModelProviders;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.widget.RecyclerView;

import com.akay.fitnass.R;
import com.akay.fitnass.data.model.Runs;
import com.akay.fitnass.view.adapters.LapAdapter;
import com.akay.fitnass.view.detail.WorkoutDetailActivity;
import com.akay.fitnass.viewmodel.DetailViewModel;

import butterknife.BindView;

public class DetailActivity extends BaseActivity {
    @BindView(R.id.recycler_workout) RecyclerView mRecyclerWorkoutLap;

    public static final String RUNS_ID_KEY = "com.akay.fitnass.ui.detail.WORKOUT_ID_KEY";

    private DetailViewModel mViewModel;
    private LapAdapter mAdapter;

    public static Intent getIntent(Context context, long idWorkout) {
        Intent intent = new Intent(context, WorkoutDetailActivity.class);
        intent.putExtra(RUNS_ID_KEY, idWorkout);
        return intent;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);
        mViewModel = ViewModelProviders.of(this).get(DetailViewModel.class);
        Runs runs = mViewModel.getById(getRunsId());
        mAdapter = new LapAdapter(runs.getLaps());
        mRecyclerWorkoutLap.setAdapter(mAdapter);
    }

    private long getRunsId() {
        Intent intent = getIntent();
        if (intent == null || !intent.hasExtra(RUNS_ID_KEY)) {
            throw new IllegalArgumentException("Workout ID is missing!");
        }
        return intent.getLongExtra(RUNS_ID_KEY, -1);
    }
}

File: app/src/main/java/com/akay/fitnass/view/activities/DetailActivity.kt
package com.akay.fitnass.view.activities

import android.arch.lifecycle.ViewModelProviders
import android.content.Context
import android.content.Intent
import android.os.Bundle
import com.akay.fitnass.R
import com.akay.fitnass.data.model.Lap
import com.akay.fitnass.view.adapters.LapAdapter
import com.akay.fitnass.viewmodel.DetailViewModel
import kotlinx.android.synthetic.main.activity_detail.*

class DetailActivity : BaseActivity() {

    private var mViewModel: DetailViewModel? = null
    private var mAdapter: LapAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)
        mViewModel = ViewModelProviders.of(this).get(DetailViewModel::class.java)
        val runs = mViewModel!!.getById(getRunsId())
        mAdapter = LapAdapter(runs.laps as MutableList<Lap>)
        recycler_workout!!.adapter = mAdapter
    }

    private fun getRunsId(): Long {
        val intent = intent
        if (intent == null || !intent.hasExtra(RUNS_ID_KEY)) {
            throw IllegalArgumentException("Workout ID is missing!")
        }
        return intent.getLongExtra(RUNS_ID_KEY, -1)
    }

    companion object {
        private const val RUNS_ID_KEY = "com.akay.fitnass.ui.detail.WORKOUT_ID_KEY"

        fun getIntent(context: Context, idWorkout: Long): Intent {
            val intent = Intent(context, DetailActivity::class.java)
            intent.putExtra(RUNS_ID_KEY, idWorkout)
            return intent
        }
    }
}
